src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.commons.io
+
+import java.nio.file.Path
+
+import com.fulcrumgenomics.commons.util.LazyLogging
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+trait CommandLineTool extends LazyLogging {
+
+  /** The name of the executable such as Rscript or gs. */
+  val Executable: String
+
+  /** Command used for the above executable.
+    *
+    * For example, to run `Rscript --version`:
+    * {{{
+    * scala> import com.fulcrumgenomics.commons.io._
+    * scala> val TestArgs = Seq("--version")
+    * scala> val Executable = "Rscript"
+    * scala> Rscript.execArgs(Executable +: TestArgs:_*)
+    * }}}
+    * */
+  val TestArgs: Seq[String]
+
+  /** Exception class that holds onto an executable's exit/status code. */
+  case class ToolException(status: Int) extends RuntimeException {
+    override def getMessage: String = s"$Executable failed with exit code $status."
+  }
+
+  /** Executes a command and returns the stdout.
+    *
+    * @param command the command to be executed
+    */
+  def execArgs(command: String*): Try[ListBuffer[String]] = {
+    Try {
+      // Read redirected stdout to a string
+      val process  = new ProcessBuilder(command: _*).redirectErrorStream(true).start()
+      val stdout   = ListBuffer[String]()
+      val sink     = new AsyncStreamSink(process.getInputStream, s => stdout.append(s))
+      val exitCode = process.waitFor()
+
+      sink.close()
+
+      require(exitCode == 0, s"Command did not execute successfully. Exit code: $exitCode")
+
+      stdout
```

Comment:
```suggestion
      yieldAndThen(stdout)(sink.safelyClose())
```

then you don't need line 68